src/main/java/software/amazon/smithy/lsp/ext/LspLog.java

```diff
@@ -0,0 +1,50 @@
+package software.amazon.smithy.lsp.ext;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This log interface buffers the messages until the server receives information
+ * about workspace root.
+ * 
+ * This helps placing the log messages into a file in the workspace, rather than
+ * pre-defined location on filesystem.
+ */
+public class LspLog {
```

Comment:
This lazy logger is **very** useful. No longer thinking of "which vs code window will this message go to"